bin/process_partis.py

```diff
@@ -111,7 +111,17 @@ def merge_upstream_seqmeta(partis_seqmeta, upstream_seqmeta):
             dup_upstream_row = get_upstream_row(upstream_seqmeta, seqid)
             # pre-partis filtering multiplicity
             dup_multiplicity = int(dup_upstream_row.get('multiplicity', 1))
-            timepoints_dict[dup_upstream_row.get('timepoint')] += dup_multiplicity
+            if not dup_upstream_row.get('timepoint'):
+                should_not_have_timepoint = seqid in seqs_without_timepoints # stored as a variable for readability on the next line
+                if ignore_timepoint_info or should_not_have_timepoint:
```

Comment:
@eharkins I'm not so sure this is the right approach here. I'd like for the processing to be as general as possible about how it handles potentially missing data. If something is missing, I think we should just fill it in, period. Maybe we could have a "strict" mode; a flag like `--only-allow-missing-timepoints-for` or something.

I think maybe part of my reservation is that `ignore_timepoint_info` sounds overly strong; It seems like the functionality you've written here would maybe better be described as `tolerate_missing_timepoints` or some such?

Reply:
The spirit of the strong requirement of all sequences having timepoints or none having timepoints (with the exception of naives and seed sequences) came from a discussion about this with Duncan. I alluded to that conversation with the quote from him in the issue for this PR #268. Full quote:
> i guess it is kind of tricky, because while in general lots of samples won't have timepoint info, if there's a sample that *does* have timepoint info but it's somehow missing for some sequences, that's extremely bad. Well, assuming the reason it's missing is that we're losing it somewhere in our software chain.

Which I felt meant that we should not do the following:
> If something is missing, I think we should just fill it in, period.

 Maybe there is a way we can do the "If [a given field] is missing, I think we should just fill it in, period." strategy and also ensure that it's **not** the case that "we're losing [that field] somewhere in our software chain."

@psathyrella any thoughts on this? @metasoarous does the concern of "losing [data] somewhere in our software chain" seem valid / explain why we would want to have stricter requirements by default for when it is ok vs not for data to be "missing" at any point during processing?

Reply:
From my point of view it all depends on what "filling it in" consists of. A priori, for sure, I'd rather handle any missing info gracefully rather than barf and make the user do something else to their inputs. For instance say we're coloring one time point red, and the other blue -- great, anything that's missing can be grey, and the user can notice "huh wait why are some of them grey". But if we're selecting out a particular time point to color and discarding others, it's very bad if we mistakenly lose some.

Reply:
This wouldn't be discarding anything, but rather plopping in "missing-timepoint" or something for those sequences which don't have timepint (or "no-timepoint"). So in the plots there would be a color for this "fake" timepoint, and you would in theory notice that way that something was off.

Part of what I don't like is that the implementation as written here assumes that a sequence will not have a timepoint if it's either a naive or a seed (basically), but that seems an overly strict assumption to me. I could imagine other reasons you would not have a timepoint, like a synthetic sequence added to a dataset. So IMHO, better to just always gracefully handle.